File: BMI/app/src/main/java/com/example/bmi/MainActivity.java
package com.example.bmi;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void BtnBmiCalc_Clicked(View v) {
        EditText editHeight = (EditText) findViewById(R.id.editHeight);
        EditText editWeight = (EditText) findViewById(R.id.editWeight);
        TextView txtResult = (TextView) findViewById(R.id.txtResult);
        Double height = Double.parseDouble(editHeight.getText().toString());
        Double weight = Double.parseDouble(editWeight.getText().toString());
        /*
            体质指数（BMI）=体重（kg）÷身高^2（m）
            最理想的体重指数是22， BMI指数为18.5～23.9时属正常，具体分类如下：
            过轻：低于18.5
            正常：18.5-23.9
            过重：24-27
            肥胖：28-32
            非常肥胖, 高于32
        */
        Double bmi = weight / (height * height);
        RadioButton rbtnStdWho = (RadioButton) findViewById(R.id.rbtnStdWho);
        RadioButton rbtnStdAsia = (RadioButton) findViewById(R.id.rbtnStdAsia);
        RadioButton rbtnStdChn = (RadioButton) findViewById(R.id.rbtnStdChn);
        if (rbtnStdWho.isChecked()) {
            if (bmi < 18.5) {
                txtResult.setText("BMI:" + bmi.toString() + " 体重：过轻 ");
            }
            else if (bmi <= 24.9) {
                txtResult.setText("BMI:" + bmi.toString() + "体重： 正常");
            }
            else if (bmi <= 29.9) {
                txtResult.setText("BMI:" + bmi.toString() + "体重： 过重");
            }
            else if (bmi <= 34.9) {
                txtResult.setText("BMI:" + bmi.toString() + "体重： 肥胖");
            }
            else {
                txtResult.setText("BMI:" + bmi.toString() + "体重： 非常肥胖");
            }
        }
        else if (rbtnStdAsia.isChecked()) {
            if (bmi < 18.5) {
                txtResult.setText("BMI:" + bmi.toString() + " 体重：过轻 ");
            }
            else if (bmi <= 22.9) {
                txtResult.setText("BMI:" + bmi.toString() + "体重： 正常");
            }
            else if (bmi <= 24.9) {
                txtResult.setText("BMI:" + bmi.toString() + "体重： 过重");
            }
            else if (bmi <= 29.9) {
                txtResult.setText("BMI:" + bmi.toString() + "体重： 肥胖");
            }
            else {
                txtResult.setText("BMI:" + bmi.toString() + "体重： 非常肥胖");
            }
        }
        else if (rbtnStdChn.isChecked()) {
            if (bmi < 18.5) {
                txtResult.setText("BMI:" + bmi.toString() + " 体重：过轻 ");
            }
            else if (bmi <= 23.9) {
                txtResult.setText("BMI:" + bmi.toString() + "体重： 正常");
            }
            else if (bmi <= 27.9) {
                txtResult.setText("BMI:" + bmi.toString() + "体重： 过重");
            }
            else {
                txtResult.setText("BMI:" + bmi.toString() + "体重： 非常肥胖");
            }
        }
        else {
            txtResult.setText("请选择BMI标准");
        }
    }
}

File: Stopwatch/app/src/main/java/bjfu/it/mukaikai/stopwatch/MainActivity.kt
package bjfu.it.mukaikai.stopwatch

import android.app.Activity
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.PersistableBundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.TextView
import java.util.*
import kotlin.math.log

class MainActivity : AppCompatActivity(), View.OnClickListener {

    var seconds:Int = 0
    var running:Boolean = false
    var wasRunning:Boolean = false
    val TAG:String = "life cycle"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.d(TAG, "onCreate")
        if (savedInstanceState != null) {
            seconds = savedInstanceState.getInt("seconds")
            running = savedInstanceState.getBoolean("running")
            wasRunning = savedInstanceState.getBoolean("wasRunning")
        }
        val btnStart:Button = findViewById(R.id.button_start)
        val btnStop:Button = findViewById(R.id.button_stop)
        val btnReset:Button = findViewById(R.id.button_reset)
        btnStart.setOnClickListener(this)
        btnStop.setOnClickListener(this)
        btnReset.setOnClickListener(this)
        runTimer()
    }
    fun runTimer() {
        var handler:Handler = object : Handler() {}
        val timeView:TextView = findViewById(R.id.time_view)
        //每隔1000ms，重复提交该任务
        handler.post(object : Runnable{
            override fun run() {
                var hours:Int = seconds/3600
                var minutes:Int = (seconds%3600)/60
                var secs:Int = seconds%60
                var time:String = String.format("%d:%02d:%02d", hours, minutes, secs)
                timeView.setText(time)
                if (running) {
                    seconds++
                }
                handler.postDelayed(this, 1000)
            }
        })
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "onStart")
        if (wasRunning) {
            running = true
        }
    }

    override fun onStop() {
        super.onStop()
        Log.d(TAG, "onStop")
        running = false
    }

    override fun onResume() {
        super.onResume()
        Log.d(TAG, "onResume")
        if(wasRunning) {
            running = true
        }
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "onPause")
        wasRunning = running
        running = false
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        super.onSaveInstanceState(outState)
        outState?.putInt("seconds", seconds)
        outState?.putBoolean("running", running)
        outState?.putBoolean("wasRunning", wasRunning)
    }

    override fun onClick(p0: View?) {
        when(p0?.id) {
            R.id.button_start -> {
                running = true
            }
            R.id.button_stop -> {
                running = false
            }
            R.id.button_reset -> {
                running = false
                seconds = 0
            }
            else -> return
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(TAG, "onDestroy")
    }
}
